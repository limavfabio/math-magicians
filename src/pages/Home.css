.home {
  margin: 3rem;
}

h1 {
  text-align: center;
  margin-bottom: 2rem;
  font-size: 2rem;
}
