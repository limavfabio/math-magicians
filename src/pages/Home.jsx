import React from 'react';
import './Home.css';

const Home = () => (
  <>
    <section className="home">
      <h1>Welcome to our page!</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet
        consequuntur possimus, nulla id praesentium porro voluptatum laboriosam
        rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda
        exercitationem.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet
        consequuntur possimus, nulla id praesentium porro voluptatum laboriosam
        rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda
        exercitationem.
      </p>
    </section>
  </>
);

export default Home;
