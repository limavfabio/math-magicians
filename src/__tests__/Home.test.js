import { render, screen } from '@testing-library/react';
import Home from '../pages/Home';

describe('Home View', () => {
  test('Renders Footer component', () => {
    const HomeSnapshot = render(<Home />);
    expect(HomeSnapshot).toMatchSnapshot();
  });

  test('Check for title in the component', () => {
    render(<Home />);
    const homeTitle = screen.getByText('Welcome to our page!');
    expect(homeTitle).toBeInTheDocument();
  });
});
