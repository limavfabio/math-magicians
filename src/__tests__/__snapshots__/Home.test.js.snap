// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Renders Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="home"
      >
        <h1>
          Welcome to our page!
        </h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem.
        </p>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem.
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="home"
    >
      <h1>
        Welcome to our page!
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquam amet consequuntur possimus, nulla id praesentium porro voluptatum laboriosam rerum ullam ipsa. Iusto alias quas, distinctio iste voluptates assumenda exercitationem.
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
